pkg/remoteconfig/state/testingutils_test.go

```diff
@@ -188,23 +188,19 @@ func newAPMSamplingFile() apmsampling.APMSampling {
 		}},
 	}
 
-	return apmConfig
+	raw, _ := apmConfig.MarshalMsg(nil)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 Suggestion:** ‏Maybe I'm being paranoid here, but shouldn't we handle any error happening here?

Reply:
Or maybe do the same as what's been removed before and give it a buffer of the right size instead of `nil`. I'm no expert in msgpack though, so feel free to ignore

Reply:
That's testing stuff so I don't personally think the overhead of checking the error is worth it but that is indeed debatable

Reply:
> give it a buffer of the right size instead of nil

Don't know if I'm understanding right but in case: if you pass nil to MarshalMsg, it does the right allocation itself IIRC